### PR TITLE
Add 0-argument anonymous function example to manual/functions

### LINKED ISSUE #32785: feature request: doc for anonymous function
https://docs.julialang.org/en/v1/manual/functions/

"A zero-argument anonymous function is written as ()->3." 

Can you provide an example of what this looks like? My try failed.

```
julia> map( ()->3, [1,2])
ERROR: MethodError: no method matching (::getfield(Main, Symbol("##15#16")))(::Int64)
Closest candidates are:
  #15() at REPL[20]:1
Stacktrace:
 [1] iterate at ./generator.jl:47 [inlined]
 [2] _collect(::Array{Int64,1}, ::Base.Generator{Array{Int64,1},getfield(Main, Symbol("##15#16"))}, ::Base.EltypeUnknown, ::Base.HasShape{1}) at ./array.jl:619
 [3] collect_similar(::Array{Int64,1}, ::Base.Generator{Array{Int64,1},getfield(Main, Symbol("##15#16"))}) at ./array.jl:548
 [4] map(::Function, ::Array{Int64,1}) at ./abstractarray.jl:2018
 [5] top-level scope at none:0
```

### PR DESCRIPTION
**Pull request by elalaouifaris**
To resolve #32785 I added a example in the corresponding section in the manual based on the example given the issue discussion.

### CODE REVIEW

File: doc/src/manual/functions.md [hunk 1]
```diff
@@ -210,7 +210,7 @@ The primary use for anonymous functions is passing them to functions which take
 as arguments. A classic example is [`map`](@ref), which applies a function to each value of
 an array and returns a new array containing the resulting values:
 
-```jldoctest
+```jldoctes
```
Comment (ararslan) [inline doc/src/manual/functions.md]:
```suggestion
```jldoctest
```
I assume this was a mistake?

File: doc/src/manual/functions.md [hunk 2]
```diff
@@ -236,6 +236,25 @@ A zero-argument anonymous function is written as `()->3`. The idea of a function
...
+A use case example is [`get`](@ref). 
+
+```julia
+get(dict, key) do
+    # default value calculated here
+    time()
+end
+```
+
+The code above is equivalent to calling get with an anonymous function containing the code
```
Comment (ararslan) [inline doc/src/manual/functions.md]:
```suggestion
The code above is equivalent to calling `get` with an anonymous function containing the code
```

File: doc/src/manual/functions.md [hunk 3]
```diff
@@ -236,6 +236,25 @@ A zero-argument anonymous function is written as `()->3`. The idea of a function
...
+inclosed between do and end, like so:
```
Comment (ararslan) [inline doc/src/manual/functions.md]:
```suggestion
enclosed between `do` and `end`, like so:
```

File: doc/src/manual/functions.md [hunk 4]
```diff
@@ -236,6 +236,25 @@ A zero-argument anonymous function is written as `()->3`. The idea of a function
...
+
+```julia
+get( () -> time(), dict, key)
```
Comment (ararslan) [inline doc/src/manual/functions.md]:
```suggestion
get(()->time(), dict, key)
```

File: doc/src/manual/functions.md [hunk 5]
```diff
@@ -236,6 +236,25 @@ A zero-argument anonymous function is written as `()->3`. The idea of a function
...
+```
+
+The call to [`time`](@ref) is delayed by wrapping it in an argumentless anonymous function
```
Comment (ararslan) [inline doc/src/manual/functions.md]:
```suggestion
The call to [`time`](@ref) is delayed by wrapping it in a 0-argument anonymous function
```

File: doc/src/manual/functions.md [hunk 6]
```diff
@@ -236,6 +236,25 @@ A zero-argument anonymous function is written as `()->3`. The idea of a function
...
+that is called only when the requested key is absent from dict.
```
Comment (ararslan) [inline doc/src/manual/functions.md]:
```suggestion
that is called only when the requested key is absent from `dict`.
```

File: doc/src/manual/functions.md [hunk 7]
```diff
@@ -236,6 +236,25 @@ A zero-argument anonymous function is written as `()->3`. The idea of a function
 may seem strange, but is useful for "delaying" a computation. In this usage, a block of code is
 wrapped in a zero-argument function, which is later invoked by calling it as `f`.
 
+A use case example is [`get`](@ref).
```
Comment (ararslan) [inline doc/src/manual/functions.md]:
```suggestion
As an example, consider this call to [`get`](@ref):
```

**Review by ararslan — CHANGES_REQUESTED** [review]
Welcome and thanks for the contribution! I have a few stylistic and wording suggestions but overall this looks good.

File: doc/src/manual/functions.md (hunk 1 above)
Comment (elalaouifaris) [reply doc/src/manual/functions.md]: Yes I noticed it after submitting.

Comment (KristofferC) [reply doc/src/manual/functions.md]: This is not resolved.

**Review by ararslan — APPROVED** [review]

**Comment by fredrikekre** [comment]
Thanks, and welcome as a contributor to Julia!

**Pull request merged**